lib/elixir/src/elixir_module.erl

```diff
@@ -32,13 +31,17 @@ defs_table(Module) ->
 is_open(Module) ->
   ets:lookup(elixir_modules, Module) /= [].
 
-delete_doc(#{module := Module}, _, _, _, _, _) ->
-  ets:delete(data_table(Module), doc),
-  ok.
+delete_doc(Data) ->
+  ets:delete(Data, doc).
 
-delete_impl(#{module := Module}, _, _, _, _, _) ->
-  ets:delete(data_table(Module), impl),
-  ok.
+delete_deprecated(Data) ->
+  ets:delete(Data, deprecated).
+
+get_deprecated(Data) ->
+  lists:usort(ets:select(Data, [{{{deprecated, '$1'}, '$2'}, [], [{{'$1', '$2'}}]}])).
```

Comment:
So to clarify the comment in the main thread, I prefer this as it was, happening in `elixir_erl`, because we don't want to store the deprecated stuff in three different places in the chunk.